src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/transform.py

```diff
@@ -53,32 +72,186 @@ def transform_update_setter(client, resource_group_name,
 
 
-def update_transform(instance, description=None, preset_names=None):
+def update_transform(instance, description=None, preset_names=None, custom_preset_path=None):
     if not instance:
         raise CLIError('The transform resource was not found.')
 
     if description:
         instance.description = description
 
-    if preset_names:
+    if preset_names or custom_preset_path:
         instance.outputs = []
+
+    if preset_names:
         for preset in preset_names:
             instance.outputs.append(get_transform_output(preset))
 
+    if custom_preset_path:
+        standard_encoder_preset = parse_standard_encoder_preset(custom_preset_path)
+        instance.outputs.append(TransformOutput(preset=standard_encoder_preset))
+
     return instance
 
 
 def get_transform_output(preset):
-    from azure.mgmt.media.models import (TransformOutput, BuiltInStandardEncoderPreset, VideoAnalyzerPreset, AudioAnalyzerPreset)
-    from azure.cli.command_modules.ams._completers import (get_stand_alone_presets, is_audio_analyzer)
+    transform_preset = None
 
-    if preset in get_stand_alone_presets():
-        if is_audio_analyzer(preset_name=preset):
-            transform_preset = AudioAnalyzerPreset()
+    try:
+        if os.path.exists(preset):
+            transform_preset = parse_standard_encoder_preset(preset)
         else:
-            transform_preset = VideoAnalyzerPreset()
-    else:
-        transform_preset = BuiltInStandardEncoderPreset(preset_name=preset)
+            if preset in get_stand_alone_presets():
+                transform_preset = get_sdk_model_class("{}Preset".format(preset))()
+            else:
+                transform_preset = BuiltInStandardEncoderPreset(preset_name=preset)
+
+    except:
+        raise CLIError("Couldn't create a preset from '{}'.".format(preset))
 
     transform_output = TransformOutput(preset=transform_preset)
     return transform_output
+
+
+def map_codecs(codecs):
+    def get_layers(layer_type, layers_list_from_json):
+        layers = []
+        for layer_group in layers_list_from_json:
+            layer_instance = get_sdk_model_class(layer_type)()
+            for layer_prop_key, layer_prop_value in layer_group.items():
+                parse_custom_preset_value(layer_prop_key, layer_prop_value, layer_instance)
+            layers.append(layer_instance)
+        return layers
+
+    codec_instance_list = []
+    for codec in codecs:
+        try:
+            codec_instance = get_sdk_model_class(map_codec_type(codec['Type']))()
+            layers = []
+            for key, value in codec.items():
+                if isinstance(value, list) and 'Layers' in key:
+                    layer_type = key[:-1]
+                    layers = get_layers(layer_type, value)
+                else:
+                    parse_custom_preset_value(key, value, codec_instance)
+        except Exception:
+            raise CLIError("Invalid or unsupported codec type in your custom preset JSON file.")
```

Comment:
I would have written "**There is an** invalid or unsupported codec type in your custom preset JSON file.", but it's ok like this too.